test/test-util.js

```diff
@@ -236,7 +193,7 @@ exports["test computeSourceURL"] = function(assert) {
   assert.equal(libUtil.computeSourceURL("src/", "test.js", "http://example.com"),
                "http://example.com/src/test.js");
   assert.equal(libUtil.computeSourceURL("src", "/test.js", "http://example.com"),
-               "http://example.com/src/test.js");
+               "http://example.com/test.js");
```

Comment:
I think this change is wrong according to the spec.  It says:

> Line 4: An optional source root, useful for relocating source files on a server or removing repeated values in the “sources” entry.  This value is prepended to the individual entries in the “source” field.

Then later:

> If the sources are not absolute URLs after prepending of the “sourceRoot”, the sources are resolved relative to the SourceMap (like resolving script src in a html document).

That is, the source root and the source are joined with string concatenation; then URL resolution is done using the source map's URL.  Here, IIUC, the source root is `src` and the source is `/test.js`, so by my reading the test was correct before this change.

Reply:
This is where I'm on the fence too. It depends on if we want to interpret `prepend`. Doing a straight string concatenation seems terrible, but it's hard to see the existing logic of adding a `/` if there isn't one as anything but super confusing spec-wise too, since the spec also never mentions prepending with a `/` when needed. So if we accept that it's supposed to be a url-like concatenation, then an absolute path does override the value in the root.

I'm find reverting this behavior, but it does seem like this would be the ideal behavior if possible.

What I kind of want to do is land this in Nightly and see if we get complaints about broken sourcemaps, but I'm not sure if that's a good approach or what. I have a hard time thinking of a case where someone's going to have a `sources` entry starting with `/` that isn't an absolute path, but you're not wrong that it could technically exist.

Reply:
I guess my feeling on it is that the spec is extremely under-specified, and as maintainers of this library we should use it to shepard the spec into a clearly-defined state.

Reply:
The spec leaves a lot to be desired, and this was a questionable choice, but I don't think this one is ambiguous.  I think if you dig through bugzilla (look at the closed bugs that block the source-maps bug) you'll find some real-world cases that motivated changes here.  I suggest doing this, and also looking through the [source map examples](https://tromey.github.io/source-map-examples/) page, before making breaking changes.  Also cross-testing against Chrome is useful, as Chrome's source map support is generally better than Firefox's.

Reply:
Fair. I see discussions in https://bugzilla.mozilla.org/show_bug.cgi?id=977463 and Lydell pretty much stating exactly what I'd have wanted behavior-wise in https://github.com/mozilla/source-map/issues/62#issuecomment-36428814. Alas :( At least I know I'm not alone in finding the behavior frustrating.

I'm going to add a change to this PR to see if we can find a good intermediate.